py4j-python/src/py4j/java_gateway.py

```diff
@@ -794,7 +936,32 @@ def __call__(self, *args):
         return return_value
 
 
-class JavaPackage():
+class UserHelpAutoCompletion(object):
```

Comment:
I'm not 100% sure what the purpose of this class is. Is this a pattern/common use case for autocompletion in IPython or ?